R/utils.R

```diff
@@ -1300,3 +1300,5 @@ setMethod("reduce", "data.frame",
     parents | children
 }
 
+windowIndices <- function(i, hws, n) max(1L, i - hws):min(n, i + hws)
```

Comment:
Should we add something like `stopifnot(i <= n)` here? Otherwise this could happen:
```r
windowIndices(i=10, hws=2, n=5)
# [1] 8 7 6 5 
```

Reply:
Makes sense. I'll do.